superset/utils/log.py

```diff
@@ -83,6 +86,43 @@ def stats_logger(self):
         return current_app.config.get("STATS_LOGGER")
 
 
+def get_event_logger_from_cfg_value(cfg_value: object) -> AbstractEventLogger:
+    """
+    This function implements the deprecation of assignment of class objects to EVENT_LOGGER
+    configuration, and validates type of configured loggers.
+
+    The motivation for this method is to gracefully deprecate the ability to configure
+    EVENT_LOGGER with a class type, in favor of preconfigured instances which may have
+    required construction-time injection of proprietary or locally-defined dependencies.
+
+    :param cfg_value: The configured EVENT_LOGGER value to be validated
+    :return: if cfg_value is a class type, will return a new instance created using a
+    default con
+    """
+    result: Any = cfg_value
+    if inspect.isclass(cfg_value):
+        logging.getLogger().warning(
```

Comment:
We could also use https://docs.python.org/3/library/exceptions.html#DeprecationWarning here. TBH I'm not super familiar with the difference or the best approach.

Reply:
I feel like this is probably a more "correct" approach, but I tried to use `warnings.warn(msg, DeprecationWarning)` and didn't see any message locally. I'm sure it can be configured, but I wanted to be sure that this would be seen.

Reply:
Other than this one (which I am open to, but it seems... extra) I implemented all of your suggestions @betodealmeida